File: .agents/sentinel/get_modified_files.js
const fs = require('fs');
const path = require('path');

const excludeDirs = new Set(['.git', '.agents', 'node_modules', '.next']);
const files = [];

function walk(dir) {
  let list;
  try {
    list = fs.readdirSync(dir);
  } catch (err) {
    return;
  }
  for (const file of list) {
    const fullPath = path.join(dir, file);
    let stat;
    try {
      stat = fs.statSync(fullPath);
    } catch (err) {
      continue;
    }
    if (stat.isDirectory()) {
      if (!excludeDirs.has(file)) {
        walk(fullPath);
      }
    } else {
      if (file !== 'package-lock.json') {
        files.push({ path: fullPath, mtime: stat.mtimeMs });
      }
    }
  }
}

try {
  walk('d:\\PROJECT\\AROH');
  files.sort((a, b) => b.mtime - a.mtime);
  files.slice(0, 5).forEach(f => console.log(f.path));
} catch (e) {
  console.error(e);
}

File: .agents/sentinel/get_modified_files.py
import os

exclude_dirs = {'.git', '.agents', 'node_modules', '.next'}
files = []
for root, dirs, filenames in os.walk('d:\\PROJECT\\AROH'):
    dirs[:] = [d for d in dirs if d not in exclude_dirs]
    for filename in filenames:
        if filename == 'package-lock.json':
            continue
        filepath = os.path.join(root, filename)
        try:
            mtime = os.path.getmtime(filepath)
            files.append((filepath, mtime))
        except OSError:
            pass

files.sort(key=lambda x: x[1], reverse=True)
for filepath, mtime in files[:5]:
    print(filepath)
